src/core/Akka/Actor/Props.cs

```diff
@@ -398,35 +357,47 @@ public static Props Empty
             if (newExpression == null)
                 throw new ArgumentException("The create function must be a 'new T (args)' expression");
 
-            object[] args = newExpression.GetArguments();
+            var args = newExpression.GetArguments();
 
-            return new Props(typeof (TActor), supervisorStrategy, args);
+            return new Props(typeof(TActor), supervisorStrategy, args);
         }
 
         /// <summary>
-        /// Creates an actor using the given arguments.
+        ///     Creates an actor using the given arguments.
         /// </summary>
         /// <typeparam name="TActor">The type of the actor to create.</typeparam>
         /// <param name="args">The arguments needed to create the actor.</param>
         /// <returns>The newly created <see cref="Akka.Actor.Props" />.</returns>
         public static Props Create<TActor>(params object[] args) where TActor : ActorBase
         {
-            return new Props(typeof(TActor), args);
+            return new Props(new ActivatorProducer(typeof(TActor), args), DefaultDeploy, args);
         }
 
         /// <summary>
-        /// Creates an actor using a specified actor producer.
+        ///     Creates an actor using a specified actor producer.
         /// </summary>
         /// <typeparam name="TProducer">The type of producer used to create the actor.</typeparam>
         /// <param name="args">The arguments needed to create the actor.</param>
         /// <returns>The newly created <see cref="Akka.Actor.Props" />.</returns>
+        [Obsolete("Do not use this method. Call CreateBy(IIndirectActorProducer, params object[] args) instead")]
         public static Props CreateBy<TProducer>(params object[] args) where TProducer : class, IIndirectActorProducer
         {
             return new Props(typeof(TProducer), args);
         }
 
         /// <summary>
-        /// Creates an actor using a specified supervisor strategy.
+        ///     Creates an actor using a specified actor producer.
+        /// </summary>
+        /// <param name="producer">The actor producer that will be used to create the underlying actor..</param>
+        /// <param name="args">The arguments needed to create the actor.</param>
+        /// <returns>The newly created <see cref="Akka.Actor.Props" />.</returns>
+        public static Props CreateBy(IIndirectActorProducer producer, params object[] args)
+        {
+            return new Props(producer, DefaultDeploy, args);
```

Comment:
even though the args presumably get passed intro the `IIndirectActorProducer`, we still have to pass them into the `Props` constructor because they're used for things like remote deployments.